lib/evil_query.rb

```diff
@@ -17,7 +17,10 @@ def initialize(user)
     end
 
     def posts
-      Post.joins(:participations).where(:participations => {:author_id => @user.person.id}).order("posts.interacted_at DESC")
+      Post.joins(:participations)
+        .where(:participations => {:author_id => @user.person.id})
+        .where("participations.count > 0")
```

Comment:
The participations table is quite big already, I think instead we should check if the counter reaches 0 while updating it and destroy the participation in that case.